frontend/leaderboard/src/Challenges.re

```diff
@@ -63,6 +65,79 @@ let getSnarkWorkCreatedByUser = blocks => {
   blocks |> calculateProperty(calculateSnarkWorkCount);
 };
 
+module SnarkFeesCollectedMap = Map.Make(String);
+
+let snarkFeesCollected = blocks => {
+  Array.fold_left(
+    (map, block: Types.NewBlock.t) => {
+      Array.fold_left(
+        (map, snarkJob: Types.NewBlock.snarkJobs) => {
+          SnarkFeesCollectedMap.update(
+            snarkJob.prover,
+            feeCount =>
+              switch (feeCount) {
+              | Some(feeCount) => Some(Int64.add(feeCount, snarkJob.fee))
+              | None => Some(snarkJob.fee)
+              },
+            map,
+          )
+        },
+        map,
+        block.data.newBlock.snarkJobs,
+      )
+    },
+    SnarkFeesCollectedMap.empty,
+    blocks,
+  );
+};
+
+let max = (a, b) => {
+  a > b ? a : b;
+};
+
+let highestSnarkFeeCollected = blocks => {
+  Array.fold_left(
+    (map, block: Types.NewBlock.t) => {
+      Array.fold_left(
+        (map, snarkJob: Types.NewBlock.snarkJobs) => {
+          SnarkFeesCollectedMap.update(
+            snarkJob.prover,
+            feeCount =>
+              switch (feeCount) {
+              | Some(feeCount) => Some(max(feeCount, snarkJob.fee))
+              | None => Some(snarkJob.fee)
+              },
+            map,
+          )
+        },
+        map,
+        block.data.newBlock.snarkJobs,
+      )
+    },
+    SnarkFeesCollectedMap.empty,
+    blocks,
+  );
+};
+
+let transactionsSentToAddress = blocks => {
+  Array.fold_left(
+    (map, block: Types.NewBlock.t) => {
+      StringMap.update(
+        block.data.newBlock.transactions.feeTransfer.recipient,
```

Comment:
I think this might not be the metric we needed. We were saying "Number of transactions sent to a given address".
So for a given address, how many transactions did you send to it. For instance, the echo bot address.
So basically this metrics function could take an address as an additional parameter and use that in the calculation

Reply:
oooo, i was thinking that meant "number of times in all the transactions where this user was the recipient"... but i know what you mean,  so the function will also take a public key